includes/admin/load.php

```diff
@@ -248,7 +248,7 @@ function perflab_install_activate_plugin_callback() {
 		wp_die( esc_html__( 'Invalid plugin.', 'performance-lab' ) );
 	}
 
-	$is_plugin_installed = isset( $_GET['file'] ) && $_GET['file'];
+	$is_plugin_installed = isset( $_GET['file'] ) && sanitize_text_field( wp_unslash( $_GET['file'] ) );
```

Comment:
Since we're only checking if the file is not null and we're already sanitizing the value in line 298, adding sanitization here would be redundant.

Reply:
It seems strange that the presence of the `file` parameter is the signal that the plugin is installed or not. This is coming from core logic?

Reply:
No, I see it is in `perflab_render_plugin_card()`. 

@mukeshpanchal27 I think we can harden/simplify things by only using the slug. This will eliminate the need for the `file` query parameter, and we add additional validation to restrict the slug to only those which are found in `plugins.json`. For example:

```diff
diff --git a/includes/admin/load.php b/includes/admin/load.php
index d4a12608..c179945c 100644
--- a/includes/admin/load.php
+++ b/includes/admin/load.php
@@ -247,8 +247,18 @@ function perflab_install_activate_plugin_callback() {
 	if ( ! $plugin_slug ) {
 		wp_die( esc_html__( 'Invalid plugin.', 'performance-lab' ) );
 	}
-
-	$is_plugin_installed = isset( $_GET['file'] ) && $_GET['file'];
+	// TODO: Make sure that $plugin_slug is inside of plugins.json. If not, then call wp_die().
+
+	// Check if installed and determine the plugin basename.
+	$is_plugin_installed = false;
+	$plugin_basename     = null;
+	foreach ( array_keys( get_plugins() ) as $plugin_file ) {
+		if ( strtok( $plugin_file, '/' ) === $plugin_slug ) {
+			$is_plugin_installed = true;
+			$plugin_basename     = $plugin_file; // TODO: The variable name "$plugin_basename" seems misleading. To follow core convention, it should be "$plugin_file", right?
+			break;
+		}
+	}
 
 	// Install the plugin if it is not installed yet.
 	if ( ! $is_plugin_installed ) {
@@ -294,8 +304,6 @@ function perflab_install_activate_plugin_callback() {
 
 		$plugin_file_names = array_keys( $plugins );
 		$plugin_basename   = $plugin_slug . '/' . $plugin_file_names[0];
-	} else {
-		$plugin_basename = sanitize_text_field( wp_unslash( $_GET['file'] ) );
 	}
 
 	if ( ! current_user_can( 'activate_plugin', $plugin_basename ) ) {
diff --git a/includes/admin/plugins.php b/includes/admin/plugins.php
index 1721cf41..5436cbf5 100644
--- a/includes/admin/plugins.php
+++ b/includes/admin/plugins.php
@@ -174,7 +174,6 @@ function perflab_render_plugin_card( array $plugin_data ) {
 					'action'   => 'perflab_install_activate_plugin',
 					'_wpnonce' => wp_create_nonce( 'perflab_install_activate_plugin' ),
 					'slug'     => $plugin_data['slug'],
-					'file'     => $status['file'],
 				),
 				admin_url( 'options-general.php' )
 			)

```

Reply:
Thanks @mukeshpanchal27 for the guide regarding redundant changes, I remove those changes on PR.

Reply:
I've submitted my above patch as https://github.com/WordPress/performance/pull/1170